docs/guides/modals.md

```diff
@@ -0,0 +1,409 @@
+
+1. Verify requests from Slack (read [this](https://api.slack.com/docs/verifying-requests-from-slack) if unfamiliar)
+1. Parse the request body, and check if the `action_id` in a block is the one you'd like to handle
+1. [Modify/push a view via API](https://api.slack.com/surfaces/modals/using#modifying) and/or update the modal to hold the sent data as [private_metadata](https://api.slack.com/surfaces/modals/using#carrying_data_between_views)
+1. Respond with 200 OK as the acknowledgment
+
+#### view_submission requests
+
+When a modal view is submitted, you'll receive a [view_submission payload](https://api.slack.com/reference/interaction-payloads/views#view_submission). All you need to do to handle Slack requests by user interactions are:
+
+1. Verify requests from Slack (read [this](https://api.slack.com/docs/verifying-requests-from-slack) if unfamiliar)
+1. Parse the request body, and check if the `type` is `view_submission` and the `callback_id` is the one you'd like to handle
+1. Extract the form data from `view.state.values`
+1. Do whatever to do such as input validations, storing them in a database, talking to external services
+1. Respond with 200 OK as the acknowledgment by either of the followings:
+  * Sending an empty body means closing only the modal
+  * Sending a body with `response_action` (possible values are `errors`, `update`, `push`, `clear`)
+
+#### view_closed requests (only when `notify_on_close` is true)
+
+Your app can optionally receive [view_closed payloads](https://api.slack.com/reference/interaction-payloads/views#view_closed) whenever a user clicks on the Cancel or x buttons. These buttons are standard in all app modals. All you need to do to handle Slack requests by user interactions are:
+
+1. Verify requests from Slack (read [this](https://api.slack.com/docs/verifying-requests-from-slack) if unfamiliar)
+1. Parse the request body, and check if the `type` is `view_closed` and the `callback_id` is the one you'd like to handle
+1. Do whatever to do at the timing
+1. Respond with 200 OK as the acknowledgment
+
+### Modal Development Tips
+
+In general, there are a few things to know when working with modals. They would be:
+
+* You need `trigger_id` in user interaction payloads to open a modal view
+* Only the inputs in `"type": "input"` blocks will be included in `view.state.values`
+* Each input/selection in non-`"input"` typed blocks such as `"section"`, `"actions"` is sent as a `block_actions` request
+* You use `callback_id` to identify a modal, a pair of `block_id` and `action_id` to identify each input in `view.state.values`
+* You can use `view.private_metadata` to hold the internal state and/or `block_actions` results on the modal
+* You respond to `view_submission` requests with `response_action` to determine the next state of a modal
+* **views.update/push** APIs are supposed to be used when receiving `block_actions` requests in modals, NOT for `view_submission` requests
```

Comment:
**views.update/push** APIs -> **[`views.update`](https://api.slack.com/methods/views.update) and [`views.push`](https://api.slack.com/methods/views.push)** methods"